etc/format-all.sh

```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
+
+. "$(dirname "${BASH_SOURCE[0]}")/../.evergreen/init.sh"
+
+files="$(find "$LIBMONGOCRYPT_DIR/src" "$LIBMONGOCRYPT_DIR/test" -type f -name '*.c' -o -name '*.h' -o -name '*.cpp' -o -name '*.hpp')"
+# shellcheck disable=SC2206
+IFS=$'\n' files=($files)
+bash "$LIBMONGOCRYPT_DIR/etc/format.sh" \
+    --style="file:$LIBMONGOCRYPT_DIR/.clang-format" \
+    -i \
+    "$@" \
+    -- "${files[@]:?}"
```

Comment:
```suggestion
    -- "${files[@]:?}"

# TODO: remove once we can use clang-format 16 InsertNewlineAtEOF.
for file in "${files[@]}"; do
    [[ -z "$(tail -c1 "$file")" ]] || printf "\n" >>"$file"
done
```
Still suggest including a cheap EOF newline test-and-append routine in this script as a workaround for lack of `InsertNewlineAtEOF`. This currently affects 7 source files under the `test` directory relative to the current state of the PR (it seems like 12 files under the `src` directory have already been fixed relative to the target branch in prior revisions).

Reply:
I'm going to postpone that, as we'll only have to wait a few weeks for LLVM 16, and that is a relatively minor formatting change.